src/AmplifyGeofenceControl/index.ts

```diff
@@ -309,6 +313,27 @@ export class AmplifyGeofenceControl {
     Array.from(checkboxes).forEach(
       (checkbox) => (checkbox.checked = this._ui.getCheckboxAllValue())
     );
+
+    this.fitAllGeofences();
+  }
+
+  fitGeofence(geofenceId: string): void {
+    const mapBounds = this._map.getBounds();
+    const geofence = this._loadedGeofences[geofenceId];
+    geofence.geometry.polygon[0].forEach((coord) => {
+      mapBounds.extend(coord);
+    });
+    this._map.fitBounds(mapBounds);
```

Comment:
Just for my understanding, how come we don't pass the `padding` option here like we do for `fitAllGeofences`?

Reply:
Hm this was something I was testing out for a different experience. One context for this weirdness is that that leftNav is overlayed on top of the map.

### When checking individual boxes
- If we include the left padding it will always add the `200px` of left padding on every single geofence added to the map. So if keep checking and unchecking the same box it will slowly keep zooming out by adding 200px

### When checking display all boxes
- Since we only check the box one the left padding is only added on the single `display all` press

Totally open to suggestions on how to improve this behavior :)